test/framework/finalizers_helpers.go

```diff
@@ -106,6 +124,8 @@ func getObjectsWithFinalizers(ctx context.Context, proxy ClusterProxy, namespace
 		setFinalizers := obj.GetFinalizers()
 
 		if len(setFinalizers) > 0 {
+			// assert if the expected finalizers are set on the resource
+			Expect(setFinalizers).To(Equal(allFinalizerAssertions[node.Object.Kind]), "for resource type ", node.Object.Kind)
```

Comment:
```suggestion
			Expect(setFinalizers).To(Equal(allFinalizerAssertions[node.Object.Kind]), "for resource type %s", node.Object.Kind)
```

optionalDescription is handled like this for len >= 2
```go
return fmt.Sprintf(optionalDescription[0].(string), optionalDescription[1:]...) + "\n"
```

Reply:
Always thought optional arguments are concatenated